### PR TITLE
Feat: add `kclvm_cli lint` command to kclvm_cli

### PR DESCRIPTION
**Pull request by He1pa**
#### 1. Does this PR affect any open issues?(Y/N) and add issue references (e.g. "fix #123", "re #123".):

- [ ] N
- [x] Y  re #109

#### 2. What is the scope of this PR (e.g. component or file name):

main.rs

#### 3. Provide a description of the PR(e.g. more details, effects, motivations or doc link):

- [x] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Other

add `kclvm_cli lint` command to call resolver lint check and skip runtime

#### 4. Are there any breaking changes?(Y/N) and describe the breaking changes(e.g. more details, motivations or doc link):

- [x] N
- [ ] Y 

#### 5. Are there test cases for these changes?(Y/N) select and add more details, references or doc links:

- [ ] Unit test
- [ ] Integration test
- [ ] Benchmark (add benchmark stats below)
- [ ] Manual test (add detailed scripts or steps below)
- [x] Other

#### 6. Release note

Please refer to [Release Notes Language Style Guide](https://kusionstack.io/docs/governance/release-policy/) to write a quality release note.

```release-note
None
```

### CODE REVIEW

**Review by Peefy — APPROVED** [review]
LGTM

File: kclvm/src/main.rs [hunk 1]
```diff
@@ -78,3 +125,23 @@ fn build_settings(matches: &ArgMatches) -> SettingsFile {
     }
     settings
 }
+
+pub fn lint_files(
```
[the remark this answers is not in the document]
Comment (He1pa) [reply kclvm/src/main.rs]: fixed

File: kclvm/src/main.rs [hunk 2]
```diff
@@ -51,6 +67,52 @@ fn main() {
...
+            (None, None) => {
+                println!("{}", matches.usage());
+            }
+            (_, _) => {
+                let mut files: Vec<&str> = match matches.values_of("INPUT") {
+                    Some(files) => files.into_iter().collect::<Vec<&str>>(),
+                    None => vec![],
+                };
+                // Config settings build
+                let settings = build_settings(&matches);
```
[the remark this answers is not in the document]
Comment (He1pa) [reply kclvm/src/main.rs:82]: move to tools crate

**Review by zong-zhe — APPROVED** [review]
LGTM

File: kclvm/tools/src/lint/mod.rs [hunk 3]
```diff
@@ -0,0 +1,25 @@
+use indexmap::IndexSet;
+use kclvm_error::{Diagnostic, Level};
+use kclvm_parser::{load_program, LoadProgramOptions};
+use kclvm_sema::resolver::resolve_program;
+
+/// Check a set of files, skips execute and divides diagnostics into error and warning
+pub fn lint_files(
```
Comment (Peefy) [inline kclvm/tools/src/lint/mod.rs:7]: More examples and tests are needed in the doc comments.

**Review by Peefy — COMMENTED** [review]
LGTM

**Review by Peefy — APPROVED** [review]
LGTM

**Review by zong-zhe — APPROVED** [review]
LGTM

**Pull request merged**